core/src/main/java/hudson/model/JDK.java

```diff
@@ -166,11 +166,8 @@ public String getDisplayName() {
             return Jenkins.getInstance().getJDKs().toArray(new JDK[0]);
         }
 
-        // this isn't really synchronized well since the list is Hudson.jdks :(
-        public @Override synchronized void setInstallations(JDK... jdks) {
-            List<JDK> list = Jenkins.getInstance().getJDKs();
-            list.clear();
-            list.addAll(Arrays.asList(jdks));
+        public @Override void setInstallations(JDK... jdks) {
+            Jenkins.getInstance().setJDKs(Arrays.asList(jdks));
```

Comment:
If you don't expect a failure (e.g. if the method is being called when Jenkins is active and ready for the work), use <code>Jenkins.getActiveInstance()</code> (it's quite new, so the legacy code uses the old method).